### ISSUE TITLE
In BertForSequenceClassification, why is loss initialised in every forward?

### ISSUE THREAD

**Issue by BramVanroy** [huggingface/transformers#1312]
Looking at [the source](https://github.com/huggingface/pytorch-transformers/blob/master/pytorch_transformers/modeling_bert.py#L902-L910) I can see that the correct loss function is initialized in each call to forward.

https://github.com/huggingface/pytorch-transformers/blob/master/pytorch_transformers/modeling_bert.py#L902-L910

Can you explain why? Why isn't the loss function set up as part of `init()`? Is there any advantage of always re-initialising it on each forward? 

Edit: I see that you do this in other parts as well, e.g. the ReLU layer in distilbert: https://github.com/huggingface/pytorch-transformers/blob/master/pytorch_transformers/modeling_distilbert.py#L598

**Comment by AdityaSoni19031997** [comment]
Also it will be nice if the user gets to use the loss_func itself, Like currently i am using that class with slight modifications to match the pipeline with different losses rather than only CrossEntropy loss.

**Comment by BramVanroy** [comment]
You can always subclass the class, to make it your own.

Some extra information for this issue: in an issue over at pytorch, it came to light that loss functions are actually meant to be imported as functions (from nn.functional) rather than modules (from nn).

**Comment by BramVanroy** [comment]
Unstale.
